chapters/interactions-nonlinear.qmd

```diff
@@ -0,0 +1,893 @@
+```
+
+```{r}
+#| label: fig-two-class-features
+#| eval: false
+data(bivariate)
+
+trans_features <- 
+  bivariate_train %>% 
+  recipe(Class ~ ., data = bivariate_train) %>% 
+  step_YeoJohnson(A, B) %>% 
+  prep() %>% 
+  bake(new_data = NULL)
+
+raw_features <- 
+  trans_features %>% 
+  pivot_longer(c(A, B), names_to = "feature", values_to = "value") %>% 
+  mutate(feature = paste(feature, "(YJ)"))
+
+ratio_features <-
+  bivariate_train %>% 
+  mutate(
+    feature = "Log-Ratio",
+    value = log(A / B)
+  ) %>% 
+  select(feature, value, Class)
+
+pca_features <- 
+  recipe(Class ~ ., data = bivariate_train) %>% 
+  step_YeoJohnson(A, B) %>% 
+  step_normalize(A, B) %>% 
+  step_pca(A, B, num_comp = 2) %>% 
+  prep() %>% 
+  bake(new_data = NULL) %>% 
+  pivot_longer(c(PC1, PC2), names_to = "feature", values_to = "value")
+
+all_features <- 
+  bind_rows(ratio_features, pca_features) %>% 
+  mutate(
+    feature = factor(feature, levels = c("Log-Ratio", "PC1", "PC2"))
+  )
+
+all_features %>% 
+  ggplot(aes(Class, value, fill = Class)) + 
+  geom_boxplot(alpha = 1 / 2) +
+  facet_wrap(~ feature, scales = "free_y") + 
+  scale_fill_manual(values = biv_cols) +
+  labs(y = NULL)
+```
+
+It is essential to understand that these _data_ require transformation. We can put the original predictors into a model as-is. The model won’t produce an error but won’t have good predictive performance. We can induce a separation of the classes when a joint feature is used. 
+
+This aspect of feature engineering depends on the data set so it is difficult to enumerate all possible techniques. In this chapter, we’ll describe a few commonly used methods: splines, interactions, and discretization (a.k.a. binning). 
+
+## Interactions {#sec-interactions}
+
+When building models for prediction, the majority of variation in the outcome is generally explained by the cumulative effect of the important individual predictors.  For many problems, additional variation in the response can be explained by the effect of two or more predictors working in conjunction with each other.  
+
+The healthcare industry has long understood the concept of interactions among drugs for treating specific diseases [@singh2017suppressive, @mokhtari2017combination, and @altorki2021neoadjuvant].  As an example of an interaction, consider treatment for the disease non-small cell lung cancer (NSCLC).  In a recent study, patients with an advanced stage of NSCLC with an EGFR mutation were given either osimertinib alone or osimertinib in combination with traditional chemotherapy [@tragrisso2023].   Patients taking the combination treatment had a significantly longer progression-free survival time than patients taking osmertinib alone.  Hence, the interaction of the treatments is more effective than the single treatment.  To summarize, two (or more) predictors interact if their combined impact is different (less or greater) than what would be expected from the added impact of each predictor alone.
```

Comment:
@kjell-stattenacity I don't have the bib entry for `tragrisso2023`.

Reply:
Where did we originally write this example?  I just can't put my finger on it.